pkg/registry/scheduling/rest/storage_scheduling.go

```diff
@@ -89,51 +94,112 @@ func (p RESTStorageProvider) v1Storage(apiResourceConfigSource serverstorage.API
 }
 
 func (p RESTStorageProvider) PostStartHook() (string, genericapiserver.PostStartHookFunc, error) {
-	return PostStartHookName, AddSystemPriorityClasses(), nil
+	return PostStartHookName, AddSchedulingDefaults(), nil
 }
 
-func AddSystemPriorityClasses() genericapiserver.PostStartHookFunc {
+// AddSchedulingDefaults adds the default cluster critical priorityClasses and clusterResourceQuota which allows
+// unlimited number of critical pods be created in `kube-system` namespace.
+func AddSchedulingDefaults() genericapiserver.PostStartHookFunc {
 	return func(hookContext genericapiserver.PostStartHookContext) error {
 		// Adding system priority classes is important. If they fail to add, many critical system
 		// components may fail and cluster may break.
-		err := wait.Poll(1*time.Second, 30*time.Second, func() (done bool, err error) {
-			schedClientSet, err := schedulingclient.NewForConfig(hookContext.LoopbackClientConfig)
-			if err != nil {
-				utilruntime.HandleError(fmt.Errorf("unable to initialize client: %v", err))
-				return false, nil
-			}
+		if err := AddSystemPriorityClasses(hookContext); err != nil {
+			return fmt.Errorf("unable to add default system priority classes: %v", err)
+		}
+		// Add default resource quota which allows unlimited number of critical pods to be created in kube-system
+		// namespace. This is to ensure backwards compatibility with existing system where we limit critical pods
+		// to be created in `kube-system`.
+		if err := AddDefaultSystemQuota(hookContext); err != nil {
+			return fmt.Errorf("unable to add default system resource quota: %v", err)
+		}
+		return nil
+	}
+}
+
+func AddSystemPriorityClasses(hookContext genericapiserver.PostStartHookContext) error {
```

Comment:
I don't think this function needs to be public.

Reply:
Sure, I can make these functions private.

Reply:
Done..